reactive/flannel.py

```diff
@@ -245,6 +248,13 @@ def cleanup_deployment():
 
 def get_flannel_subnet():
     ''' Returns the flannel subnet reserved for this unit '''
-    with open('/run/flannel/subnet.env') as f:
-        raw_data = dict(line.strip().split('=') for line in f)
-    return raw_data['FLANNEL_SUBNET']
+    try:
+        with open('/run/flannel/subnet.env') as f:
+            raw_data = dict(line.strip().split('=') for line in f)
+        return raw_data['FLANNEL_SUBNET']
+    except FileNotFoundError as e:
+        raise FlannelSubnetNotFound() from e
```

Comment:
This is for readability sake. Just raising an exception with a name that I think is more appropriate.

So when the caller of `get_flannel_subnet()` has an exception handler for `FlannelSubnetNotFound`, it should hopefully be pretty clear where that's coming from. Whereas if they're catching `FileNotFoundError`, it may not be as obvious.